packages/@aws-cdk/aws-scheduler-alpha/lib/target.ts

```diff
@@ -0,0 +1,277 @@
+   * The templated target type for the Amazon SQS SendMessage API Operation
+   */
+  readonly sqsParameters?: CfnSchedule.SqsParametersProperty;
+}
+
+/**
+ * Bind props to base schedule target config.
+ * @internal
+ */
+export function bindBaseTargetConfig(props: ScheduleTargetBaseProps) {
+  let { deadLetterQueue } = props;
+
+  return {
+    deadLetterConfig: deadLetterQueue ? { arn: deadLetterQueue?.queueArn } : undefined,
+    retryPolicy: renderRetryPolicy(props.maxEventAge, props.retryAttempts),
+  };
+}
+
+const roleCache: {[key: string]: iam.Role} = {};
+
+/**
+ * Obtain the Role for the EventBridge Scheduler event
+ *
+ * If a role already exists, it will be returned. This ensures that if multiple
+ * events have the same target, they will share a role.
+ * @internal
+ */
+function singletonScheduleRole(schedule: ISchedule, targetArn: string): iam.IRole {
+  const stack = Stack.of(schedule);
+  const arn = Token.isUnresolved(targetArn) ? stack.resolve(targetArn).toString() : targetArn;
+  const hash = md5hash(arn).slice(0, 6);
+  const id = 'SchedulerRoleForTarget-' + hash;
+  const existing = stack.node.tryFindChild(id) as iam.IRole;
+
+  const principal = new iam.PrincipalWithConditions(new iam.ServicePrincipal('scheduler.amazonaws.com'), {
+    StringEquals: {
+      'aws:SourceAccount': schedule.env.account,
+    },
+  });
+  if (existing) {
+    roleCache[targetArn].assumeRolePolicy?.addStatements(new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      principals: [new iam.ServicePrincipal('scheduler.amazonaws.com')],
+      actions: ['sts:AssumeRole'],
+    }));
+    return existing;
+  }
+  const role = new iam.Role(stack, id, {
+    roleName: PhysicalName.GENERATE_IF_NEEDED,
+    assumedBy: principal,
+  });
+  roleCache[targetArn] = role;
+  return role;
+}
+
+/**
+ * Allow a schedule to send events with failed invocation to an Amazon SQS queue.
+ * @internal
+ */
+export function addToDeadLetterQueueResourcePolicy(schedule: ISchedule, queue: sqs.IQueue) {
+  if (!sameEnvDimension(schedule.env.region, queue.env.region)) {
+    throw new Error(`Cannot assign Dead Letter Queue in region ${queue.env.region} to the schedule ${Names.nodeUniqueId(schedule.node)} in region ${schedule.env.region}. Both the queue and the schedule must be in the same region.`);
+  }
+
+  // Skip Resource Policy creation if the Queue is not in the same account.
+  // There is no way to add a target onto an imported schedule, so we can assume we will run the following code only
+  // in the account where the schedule is created.
+  if (sameEnvDimension(schedule.env.account, queue.env.account)) {
+    const policyStatementId = `AllowSchedule${Names.nodeUniqueId(schedule.node)}`;
+
+    queue.addToResourcePolicy(new iam.PolicyStatement({
+      sid: policyStatementId,
+      principals: [new iam.ServicePrincipal('events.amazonaws.com')],
+      effect: iam.Effect.ALLOW,
+      actions: ['sqs:SendMessage'],
+      resources: [queue.queueArn],
```

Comment:
@Jacco  similarly, there was an option to create a condition for `sourceArn` here. But it would create a dependency cycle between target and schedule.